### ISSUE TITLE
before_task_start command will not run by other module call using do_task command

### ISSUE THREAD

**Issue by petersonchen** [RexOps/Rex#667]
consider below code :

== module Foo:Bar ==
```perl
package Foo::Bar;
task "uptime" , "192.168.0.10" , sub {
	say run("uptime");
};

before_task_start "uptime" => sub {
	say "before start execute uptime";
};
```
== module Foo:Main ==
```perl
package Foo::Main;
include qw/ Foo::Bar /;
task "go"  , sub {
	do_task("Foo:Bar:uptime");
};
```

==  CLI ==
```
rex -f lib/Foo/Main.pm go 
```

== expect ==
```
before start execute uptime
11:53:39 up 1 day, 19:44,  1 user,  load average: 0.10, 0.10, 0.06
```

== Actual (before_task_start  will not be executed)==
```
11:53:39 up 1 day, 19:44,  1 user,  load average: 0.10, 0.10, 0.06
```

== Note ==
1. the above code is expected on Rex V1.1.0  , but It does not work on Rex v 1.2.0
2. after_task_finished has the same problem , too.
3. before_task_start  will run normally  if execute task on module directly
     rex -f lib/Foo/Bar.pm  uptime 
    == output ==
    before start execute uptime
    11:53:39 up 1 day, 19:44,  1 user,  load average: 0.10, 0.10, 0.06

**Comment by krimdomu** [comment]
Hi @petersonchen ,

thanks for the report. I'm going to look at it. It seems that this is only happening when using `include`. As a workaround until it is fixed, you can use `use` or `require`.

```perl
package Foo::Main;
use  Foo::Bar;
task "go"  , sub {
    do_task("Foo:Bar:uptime");
};
```

This will display the tasks of `Foo::Bar` on `rex -T` but at least it will run the hooks.

**Comment by krimdomu** [comment]
@petersonchen ,
this should now be fixed with the commit ac63332.

You can download these 2 files and replace the one from your installation:

https://raw.githubusercontent.com/RexOps/Rex/ac63332e244b62b8f157b468fb90f8d6e29ff3c1/lib/Rex/TaskList/Base.pm
https://raw.githubusercontent.com/RexOps/Rex/ac63332e244b62b8f157b468fb90f8d6e29ff3c1/lib/Rex/CLI.pm

**Comment by krimdomu** [comment]
I'm going to write tests for this later the day.
